core/extension/gdnative_interface.h

```diff
@@ -0,0 +1,413 @@
+
+	GDNATIVE_VARIANT_TYPE_VARIANT_MAX
+};
+
+enum GDNativeVariantOperator {
+	//comparison
+	NATIVE_VARIANT_OP_EQUAL,
+	NATIVE_VARIANT_OP_NOT_EQUAL,
+	NATIVE_VARIANT_OP_LESS,
+	NATIVE_VARIANT_OP_LESS_EQUAL,
+	NATIVE_VARIANT_OP_GREATER,
+	NATIVE_VARIANT_OP_GREATER_EQUAL,
+	//mathematic
+	NATIVE_VARIANT_OP_ADD,
+	NATIVE_VARIANT_OP_SUBTRACT,
+	NATIVE_VARIANT_OP_MULTIPLY,
+	NATIVE_VARIANT_OP_DIVIDE,
+	NATIVE_VARIANT_OP_NEGATE,
+	NATIVE_VARIANT_OP_POSITIVE,
+	NATIVE_VARIANT_OP_MODULE,
+	//bitwise
+	NATIVE_VARIANT_OP_SHIFT_LEFT,
+	NATIVE_VARIANT_OP_SHIFT_RIGHT,
+	NATIVE_VARIANT_OP_BIT_AND,
+	NATIVE_VARIANT_OP_BIT_OR,
+	NATIVE_VARIANT_OP_BIT_XOR,
+	NATIVE_VARIANT_OP_BIT_NEGATE,
+	//logic
+	NATIVE_VARIANT_OP_AND,
+	NATIVE_VARIANT_OP_OR,
+	NATIVE_VARIANT_OP_XOR,
+	NATIVE_VARIANT_OP_NOT,
+	//containment
+	NATIVE_VARIANT_OP_IN,
+	NATIVE_VARIANT_OP_MAX
+
+};
+
+typedef void *GDNativeVariantPtr;
+typedef void *GDNativeStringNamePtr;
+typedef void *GDNativeStringPtr;
+typedef void *GDNativeObjectPtr;
+typedef void *GDNativeTypePtr;
+typedef void *GDNativeMethodBindPtr;
+typedef int64_t GDNativeInt;
+typedef uint32_t GDNativeBool;
+typedef uint64_t GDObjectInstanceID;
+
+/* VARIANT DATA I/O */
+
+enum GDNativeCallErrorType {
+	NATIVE_CALL_OK,
+	NATIVE_CALL_ERROR_INVALID_METHOD,
+	NATIVE_CALL_ERROR_INVALID_ARGUMENT, // expected is variant type
+	NATIVE_CALL_ERROR_TOO_MANY_ARGUMENTS, // expected is number of arguments
+	NATIVE_CALL_ERROR_TOO_FEW_ARGUMENTS, // expected is number of arguments
+	NATIVE_CALL_ERROR_INSTANCE_IS_NULL,
+
+};
+
+struct GDNativeCallError {
+	GDNativeCallErrorType error;
+	int32_t argument;
+	int32_t expected;
+};
+
+typedef void (*GDNativeVariantFromTypeConstructorFunc)(GDNativeVariantPtr, GDNativeTypePtr);
+typedef void (*GDNativeTypeFromVariantConstructorFunc)(GDNativeTypePtr, GDNativeVariantPtr);
+typedef void (*GDNativePtrOperatorEvaluator)(const GDNativeTypePtr p_left, const GDNativeTypePtr p_right, GDNativeTypePtr r_result);
+typedef void (*GDNativePtrBuiltInMethod)(GDNativeTypePtr p_base, const GDNativeTypePtr *p_args, GDNativeTypePtr r_return, int p_argument_count);
+typedef void (*GDNativePtrConstructor)(GDNativeTypePtr p_base, const GDNativeTypePtr *p_args);
+typedef void (*GDNativePtrSetter)(GDNativeTypePtr p_base, const GDNativeTypePtr p_value);
+typedef void (*GDNativePtrGetter)(const GDNativeTypePtr p_base, GDNativeTypePtr r_value);
+typedef void (*GDNativePtrIndexedSetter)(GDNativeTypePtr p_base, GDNativeInt p_index, const GDNativeTypePtr p_value);
+typedef void (*GDNativePtrIndexedGetter)(const GDNativeTypePtr p_base, GDNativeInt p_index, GDNativeTypePtr r_value);
+typedef void (*GDNativePtrKeyedSetter)(GDNativeTypePtr p_base, const GDNativeTypePtr p_key, const GDNativeTypePtr p_value);
+typedef void (*GDNativePtrKeyedGetter)(const GDNativeTypePtr p_base, const GDNativeTypePtr p_key, GDNativeTypePtr r_value);
+typedef bool (*GDNativePtrKeyedChecker)(const GDNativeVariantPtr p_base, const GDNativeVariantPtr p_key);
```

Comment:
This is using `bool`. So either it should include `<stdbool.h>` or change to `GDNativeBool`.

Reply:
good catch

Reply:
Which would be best? `stdbool.h` is C99 IINM, so I guess if we use it (like we do so far in GDNative) it means we're not compatible with C89. Do we want to be? If we're fine being C99+ then I'd say including `stdbool.h` and using `bool` like in a proper modern language sounds best.

Reply:
changed to uint32_t